src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -171,6 +171,7 @@ bool ShenandoahConcurrentGC::collect(GCCause::Cause cause) {
   entry_cleanup_early();
 
   {
+    // TODO: Not sure there is value in logging free-set status right here
```

Comment:
Doesn't early cleanup rebuild the freeset?

Reply:
I'm not sure.  It does recycle immediate trash.  But it appears to me that the rebuild of freeset following the recycling of trash happens somewhere else.  Of note, the last thing done at the end of each free set rebuild is to log the status, so even if we do rebuild inside entry_cleanup_early(), there would be no need to log the status again right here.

Of course, one reason I left this as a TODO comment instead of just yanking out the code is because it is a bit difficult to understand the control flows, so I thought this could deserve some additional attention at a later time.

Reply:
Okay.